LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs

```diff
@@ -71,23 +71,11 @@ internal static class DownlinkMessageBuilder
             {
                 tmst = rxpk.Tmst + timeWatcher.GetReceiveWindow2Delay(loRaDevice) * 1000000;
 
-                if (string.IsNullOrEmpty(configuration.Rx2DataRate))
-                {
-                    Logger.Log(loRaDevice.DevEUI, "using standard second receive windows", LogLevel.Information);
-                    freq = loRaRegion.RX2DefaultReceiveWindows.frequency;
-                    datr = loRaRegion.DRtoConfiguration[loRaRegion.RX2DefaultReceiveWindows.dr].configuration;
-                }
-                else
-                {
-                    // if specific twins are set, specify second channel to be as specified
-                    freq = configuration.Rx2DataFrequency;
-                    datr = configuration.Rx2DataRate;
-                    Logger.Log(loRaDevice.DevEUI, $"using custom DR second receive windows freq : {freq}, datr:{datr}", LogLevel.Information);
-                }
+                (freq, datr) = request.Rxpk.GetDownstreamRX2DRAndFreq(loRaDevice.DevEUI, configuration.Rx2DataRate, configuration.Rx2DataFrequency, loRaRegion, loRaDevice.ReportedRX2DataRate);
```

Comment:
can you add a comment explaining why the calculation for the RX2 has to go through the Rxpk vs. the calculation below for the RX1? To me that's not obvious.

Reply:
I am not sure I follow you is your question asking why I did not put the method in the LoRaRegion and in the Rxpk instead?

Reply:
basically, I'm just saying, I don't understand what's going on when looking at the code and suggest to add a comment to make it clear why the 2 windows take such different routes when it comes to the freq / datr calculation.